apps/explorer/src/ui/SenderRecipient.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CoinBalance } from '~/ui/CoinBalance';
+import { Heading } from '~/ui/Heading';
+import { SenderRecipientAddress } from '~/ui/SenderRecipientAddress';
+
+type Recipient = {
+    address: string;
+    coin?: {
+        amount: number | string;
+        symbol?: string;
+    };
+};
+
+export interface SenderRecipientProps {
+    sender: string;
+    transferCoin?: boolean;
+    recipients?: Recipient[];
+}
+
+export function SenderRecipient({
+    sender,
+    recipients,
+    transferCoin,
+}: SenderRecipientProps) {
+    const multipleRecipients = recipients && recipients?.length > 1;
+    const singleTransferCoin = !!(
+        !multipleRecipients &&
+        transferCoin &&
+        recipients?.length
+    );
+    const primaryRecipient = singleTransferCoin && recipients?.[0];
+    const multipleRecipientsList = primaryRecipient
+        ? recipients?.filter((_, i) => i !== 0)
+        : recipients;
+
+    return (
+        <div className="flex flex-col justify-start text-sui-grey-100 gap-4">
+            <Heading variant="heading4" weight="semibold">
+                Sender {singleTransferCoin && '& Recipient'}
```

Comment:
Not something you need to do now, but in general we should get into the habit of making copy like this full strings we differentiate:
```tsx
{singleTransferCoin ? 'Sender' : 'Sender & Recipient'}
```

Reply:
Noted.